src/frontend/components/ViewersList/index.spec.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+
+import {
+  participantMockFactory,
+  videoMockFactory,
+} from 'utils/tests/factories';
+import { wrapInIntlProvider } from 'utils/tests/intl';
+import { useParticipantsStore } from 'data/stores/useParticipantsStore/index';
+import { ViewersList } from '.';
+
+const mockParticipant = participantMockFactory();
+
+const mockVideo = videoMockFactory({
+  participants_in_discussion: [mockParticipant],
+});
+
+describe('<ViewersList />', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('renders the correct viewers list if the user is not an instructor', () => {
+    useParticipantsStore.setState({
+      participants: [
+        {
+          ...mockParticipant,
+          isInstructor: false,
+          isOnStage: false,
+        },
+      ],
+    });
+
+    render(
+      wrapInIntlProvider(
+        <ViewersList video={mockVideo} isInstructor={false} />,
+      ),
+    );
+
+    screen.getByText(mockParticipant.name);
+  });
+
+  it('renders the correct viewers list if the user is an instructor', () => {
+    useParticipantsStore.setState({
+      participants: [
+        {
+          ...mockParticipant,
+          isInstructor: false,
+          isOnStage: false,
+        },
+      ],
+    });
+
+    render(
+      wrapInIntlProvider(<ViewersList video={mockVideo} isInstructor={true} />),
+    );
+
+    screen.getByText(mockParticipant.name);
```

Comment:
I don't see the difference between this test and the previous one. What should change ?

Reply:
I refactored this test to make it clearer in commit :  [ddfe066](https://github.com/openfun/marsha/pull/1443/commits/ddfe0664693d95bb02a75a0a7cd91842cedf6661)